### ISSUE TITLE
Incompatible with 2020.2 IDEs

### ISSUE THREAD

**Issue by hoichi** [reasonml-editor/reasonml-idea-plugin#263]
plugin version: 0.90.1-2020.1

### Description

The new WebStorm 2020.2 says:

> **Plugin Error:** Plugin "ReasonML" is incompatible (until build 201.SNAPSHOT < WS-202.6397.88).

The About dialog:
```
WebStorm 2020.2
Build #WS-202.6397.88, built on July 24, 2020
Runtime version: 11.0.7+10-b944.20 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1638M
Cores: 6
Registry: bigger.font.in.project.view=true, ide.balloon.shadow.size=0
Non-Bundled Plugins: PlantUML integration, com.chrisrm.idea.MaterialThemeUI, com.mallowigi, org.elm.klazuka, quokka.js
```

The above is from Windows, but it’s the same on Linux Mint 19.3.

For comparison, it works in IDEA, which is a later build: `Build #IU-202.6397.94, built on July 27, 2020`. So maybe it’s just a matter of updating the expected WS version.

**Comment by giraud** [comment]
The plugin is not compatible with 2020.2, see #259

**Comment by hoichi** [comment]
Ok, subscribed to #259 )

**Comment by unsignedint** [comment]
> For comparison, it works in IDEA, which is a later build: `Build #IU-202.6397.94, built on July 27, 2020`. So maybe it’s just a matter of updating the expected WS version.

Is that correct? It doesn't work for me on IDEA Ultimate 2020.2 (or any 2020.2 releases). Anyway, thanks @giraud for your awesome work.

**Comment by hoichi** [comment]
> Is that correct? It doesn't work for me on IDEA Ultimate 2020.2 (or any 2020.2 releases). Anyway, thanks @giraud for your awesome work.

I’ve checked once more, and the thing is, I’ve installed the plugin in IDEA _after_ it failed in WebStorm, just to check, IDEA installed the version `0.82.1`, which seems to be compatible (but also lacks all the latest features. WebStorm, on the other side, had the latest version, `0.90.1-2020.1`, because it’s already been installed. When I uninstalled and reinstalled the plugin in WS, I got the same version as in IDEA, and that version works.

**Comment by kamilkuz** [comment]
Any news here ?

**Comment by giraud** [comment]
not yet

**Comment by JVictorV** [comment]
Reinstalling the plugin fixed the problem for me

**Comment by hoichi** [comment]
> Reinstalling the plugin fixed the problem for me

Did you get the latest version after the reinstall? Or the `0.82.1`?

**Comment by JVictorV** [comment]
> > Reinstalling the plugin fixed the problem for me
> 
> 
> 
> Did you get the latest version after the reinstall? Or the `0.82.1`?
Got the 0.82.1 version 😔

**Comment by giraud** [comment]
Found what is wrong with current implementation, working on applying the fix for all symbols.

**Comment by giraud** [comment]
If someone wants to test the next plugin release made compatible with version 2020.2, use that link: https://dl.bintray.com/giraud/idea-plugin/idea-plugin-0.92-refs-beta01-2020.2.zip 

References have been rewritten, so if you test it and can tell me what is no more working that would be great. Just things that were working before and are now broken, not every problems 😄 .

you can add broken functionalities in this issue.

summary:
- rewritten parsers (including rescript) ... expect problems
- references/find usages are new code ... expect problems
- goto declaration/implementation should work 
- includes the new settings panel 

Thanks

**Comment by Coobaha** [comment]
> references/find usages are new code ... expect problems 
Go to definition to most of the external modules are not working

**Comment by giraud** [comment]
Thanks for testing

**Comment by unsignedint** [comment]
- Go to definition and cmd-click seems to work OK for me. Except for one quirk: clicking on module, e.g. "ReasonReact" jumps to a seemingly random point in the file.
- Find usages: doesn't work at all
- Some highlighting differences:
<img width="670" alt="Screen Shot 2020-09-11 at 9 58 19 AM" src="https://user-images.githubusercontent.com/929353/92813729-e1eb3480-f415-11ea-81d7-74c800adbda9.png">

But, it seems usable, so a great step forward.

**Comment by giraud** [comment]
@unsignedint Thanks.

- when you ctrl-click on a file module, you go to the last known position you were in that file. it's not different than before. I just reference the file and idea choose the location based on a previous edit
- Find usages must work for files in the source directory. it doesn't work from files inside node_modules. if not the case, do you have a specific example ?
- I removed some of the highlighting on variant because I'm not sure how to detect them. I have fixed some highlighting since.

I updated the link with a beta 02

it's strange because all goto seem to work but the inverse operation (find usage) is not

**Comment by unsignedint** [comment]
<img width="190" alt="Screen Shot 2020-09-15 at 1 11 33 PM" src="https://user-images.githubusercontent.com/929353/93153737-0b79c800-f756-11ea-9a28-5968149ec030.png">

Variant highlight added to the type declaration, but not the usage. Note: is seems optional highlighting works. I guess that is treated differently?

Re: find usages. Yes it does work fine. I'm not sure why it didn't seem to work for me before. (Though, I did invalidate my cache..)

**Comment by giraud** [comment]
updated beta to version 04.

I had to re-clone my project and restart from scratch to have find usages working correctly. invalidating caches wasn't enough for me, I don't know why.

**Comment by Coobaha** [comment]
Indeed, removing .idea helped.

**Comment by giraud** [comment]
Yeah, good to know there is a workaround. It's   annoying but I dont see how to do it automatically. 
I think beta04 is a good basis for a release

**Comment by unsignedint** [comment]
@giraud I think beta04 seems to be working well. I see the syntax highlighting of variants is working again. 👍 
BTW: I suggest editing the title of this issue to "Incompatible with IDEA 2020.2 IDEs" for completeness.

**Comment by giraud** [comment]
should be fixed in version 0.92
